src/Apps/Order/Components/AddressVerificationFlow.tsx

```diff
@@ -0,0 +1,309 @@
+import { SystemQueryRenderer } from "System/Relay/SystemQueryRenderer"
+import { createFragmentContainer, graphql } from "react-relay"
+import { AddressVerificationFlowQuery } from "__generated__/AddressVerificationFlowQuery.graphql"
+import { AddressVerificationFlow_verifiedAddressResult$data } from "__generated__/AddressVerificationFlow_verifiedAddressResult.graphql"
+import {
+  BorderedRadio,
+  Box,
+  Button,
+  Flex,
+  ModalDialog,
+  RadioGroup,
+  Spacer,
+  Text,
+} from "@artsy/palette"
+import { useCallback, useEffect, useState } from "react"
+
+interface AddressVerificationFlowProps {
+  verifiedAddressResult: AddressVerificationFlow_verifiedAddressResult$data
+  onChosenAddress: (address: AddressValues) => void
+  onClose: () => void
+}
+
+export interface AddressValues {
+  addressLine1: string
+  addressLine2?: string
+  country: string
+  city: string
+  region: string
+  postalCode: string
+}
+
+enum ModalType {
+  SUGGESTIONS = "suggestions",
+  REVIEW_AND_CONFIRM = "review_and_confirm",
+}
+
+interface AddressOption {
+  key: string
+  recommended: boolean
+  lines: string[]
+  address: AddressValues
+}
+
+enum AddressSuggestionRadioButton {
+  recommended = "Recommended",
+  user_address = "What you entered",
+}
+
+const AddressVerificationFlow: React.FC<AddressVerificationFlowProps> = ({
+  verifiedAddressResult,
+  onChosenAddress,
+  onClose,
+}) => {
+  console.log({ queryResult: verifiedAddressResult })
+  const [modalType, setModalType2] = useState<ModalType | null>(null)
+  const [addressOptions, setAddressOptions2] = useState<AddressOption[]>([])
+
+  const [selectedAddressKey, setSelectedAddressKey2] = useState<string | null>(
+    null
+  )
+
+  const setModalType = (type: ModalType | null) => {
+    console.log("modal", type)
+    setModalType2(type)
+  }
+  const setAddressOptions = (options: AddressOption[]) => {
+    console.log("options", options)
+    setAddressOptions2(options)
+  }
+  const setSelectedAddressKey = (key: string) => {
+    console.log("key", key)
+    setSelectedAddressKey2(key)
+  }
+
+  const chooseAddress = useCallback(() => {
```

Comment:
**question [non-blocking]:** 

Got a quick question, so it looks like choosing the selected address is based on string matching by address key, right? Would there be any possibility of key collisions here? Like, could two different addresses wind up with the same key property and maybe cause some unexpected behavior?

*format via [Conventional Comments](https://conventionalcomments.org/)*

Reply:
I don't think that there is a risk of key collision because each suggestion gets a unique key based on its position in the Smarty API response that we get from Gravity. In the `useEffect` hook that builds up the `addressOptions` state variable, we set the user-entered address key to `userAddress` and the suggested address keys to `suggestedAddress-0`, `suggestedAddress-1`, etc.

Reply:
Ah okay, so the addresses should stay consistent as long as the Smarty API's response order doesn't change between calls and the userAddress is assigned a different key than the suggestedAddress. Thanks for explaining this :)